src/features/generic/time-utils.ts

```diff
@@ -56,37 +56,86 @@ namespace TimeUtilities {
 
-    const billableHours = (projectTime/expected * 100).toFixed(1);
-    const billableHoursWithPercentage = `${billableHours}%`;
+    const billableHoursPercentage = (projectTime/expected * 100).toFixed(1);
 
-    const overtime = TimeUtilities.timeConversion(total);
     const undertime = TimeUtilities.timeConversion(total * -1);
+    const overtime = TimeUtilities.timeConversion(total);
 
-    let underOverMessage = "No overtime, you worked the expected time.";
-    if (total < 0) {
-      underOverMessage = `Undertime: ${undertime}`;
+    let message ="You worked the expected amount of time";
+    if (total > 0){
+      message = `Overtime: ${overtime}`;
+    } else if (total < 0){
+      message = `Undertime: ${undertime}`;
     }
     
-    if (total > 0) {
-      underOverMessage = `Overtime: ${overtime}`;
-    }
-
     return {
-      underOverMessage: underOverMessage,
-      billableHoursWithPercentage: billableHoursWithPercentage,
-      billableHours: billableHours
+      message: message,
+      billableHoursPercentage: billableHoursPercentage
     };
   };
+
+  /**
+   * 
+   * @param timeRegistrations All timeregistrations after the day before yesterday
+   * @param personId Users id
+   * @param expected Users expected amount of work
+   * @param date Today either yesterday depending on if function is checking is user on vacation or is it first day back at work
+   * @returns Undefined if can't find a time registration
+   */
+  export const checkIfUserIsAwayOrIsItFirstDayBack = (
+    timeRegistrations: TimeRegistrations[],
+    personId: number,
+    expected: number,
+    date: string) => {
+    return timeRegistrations.find(
+      timeRegistration => timeRegistration.person === personId
+      && timeRegistration.date === date
+      && timeRegistration.time_registered === expected);
+  };
+
+  /**
+   * 
+   * @returns yesterday, day before yesterday and number of today
+   */
+  export const yesterdayAndDayBeforeYesterdayDateProvider = () => {
+    const today = DateTime.now().toISODate();
+
+    let yesterday = DateTime.now().minus({ days: 1 }).toISODate();
+    let dayBeforeYesterday = DateTime.now().minus({ days: 2 }).toISODate();
+    
+    const dayOfWeek = new Date().getDay();
+
+    if (dayOfWeek === 1) {
+      yesterday = DateTime.now().minus({ days: 3 }).toISODate();
+      dayBeforeYesterday = DateTime.now().minus({ days: 4 }).toISODate();
+
+      const yesterdayAndDayOfWeek: YesterdayAndDayBeforeYesterdayDates = {
+        today: today,
+        yesterday: yesterday,
+        numberOfToday: dayOfWeek,
+        dayBeforeYesterday: dayBeforeYesterday
+      };
+
+      return yesterdayAndDayOfWeek;
+    } else {
+      const yesterdayAndDayOfWeek:YesterdayAndDayBeforeYesterdayDates = {
+        today: today,
+        yesterday: yesterday,
+        numberOfToday: dayOfWeek,
+        dayBeforeYesterday: dayBeforeYesterday
+      };
+
+      return yesterdayAndDayOfWeek;
+    }
+  };
```

Comment:
This seems a bit too over complicated and the names are a bit misleading. The goal is to get two previous work days so I'd change this to something like:
```
  /**
   * Gets two previous workdays
   *
   * @returns two previous workdays
   */
  export const getPreviousTwoWorkdays = (): YesterdayAndDayBeforeYesterdayDates => {
    const today = DateTime.now();
    const dayOfWeek = new Date().getDay();

    let previousWorkDay = today.minus({ days: 1 }).toISODate();
    let dayBeforePreviousWorkDay = today.minus({ days: 2 }).toISODate();

    if (dayOfWeek === 1) {
      previousWorkDay = today.minus({ days: 3 }).toISODate();
      dayBeforePreviousWorkDay = today.minus({ days: 4 }).toISODate();
    }

    return {
      today: today.toISODate(),
      yesterday: previousWorkDay,
      numberOfToday: dayOfWeek,
      dayBeforeYesterday: dayBeforePreviousWorkDay
    };
  };
```

Reply:
Done